capa/main.py

```diff
@@ -351,13 +353,131 @@ def pbar(s, *args, **kwargs):
     return matches, meta
 
 
+def find_thread_capabilities(
+    ruleset: RuleSet, extractor: DynamicFeatureExtractor, ph: ProcessHandle, th: ThreadHandle
+) -> Tuple[FeatureSet, MatchResults]:
+    """
+    find matches for the given rules for the given thread.
+
+    returns: tuple containing (features for thread, match results for thread)
+    """
+    # all features found for the thread.
+    features = collections.defaultdict(set)  # type: FeatureSet
```

Comment:
can this be `features: FeatureSet = ...`?

Reply:
yes. I've added that now.

Reply:
uh, for some reason doing type annotations like that here raises a syntax error for mypy, and that's why the type annotation is done using the `# type: FeatureSet` comment. I've added that to other similar instances.

Reply:
ok, I figured it was either there for a reason or legacy, let's add a note on it to clarify